app/components/primer/beta/spinner.html.erb

```diff
@@ -1,7 +1,9 @@
-<%= render Primer::BaseComponent.new(**@system_arguments) do %>
-  <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke" fill="none" />
-  <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-width="2" stroke-linecap="round" vector-effect="non-scaling-stroke" />
-<% end %>
-<% if no_aria_label? %>
-  <span class="spinner-screenreader-text sr-only"><%= @sr_text %></span>
+<%= render(Primer::BaseComponent.new(tag: :span, hidden: @hidden, data: { target: @target })) do %>
```

Comment:
aaah, I'm guessing the `Primer::BaseComponent.new()` is the magic bit I needed to be able to access the screenreader-only element from other spinner files?

Reply:
Sort of, we just needed to wrap both the `<svg>` and the `<span class="sr-only">` in another `<span>` so we could hide them both together. It was a little tricky because usually we pass `**system_arguments` to the top-level element (eg. the new `<span>`), but that would break the existing API which puts `**system_arguments` on the `<svg>` element. I had to extract only certain attributes, eg. `data-target` and `hidden`, to be able to put them on the new `<span>` and pass everything else down to the `<svg>`.